task-2/src/main/java/com/ury/MyHashMap.java

```diff
@@ -0,0 +1,167 @@
+package com.ury;
+
+public class MyHashMap<K, V> {
+
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private int initialCapacity;
+    private float loadFactor;
+    private int size = 0;
+    public Node<K, V>[] buckets;
+
+    public MyHashMap() {
+        initialCapacity = DEFAULT_INITIAL_CAPACITY;
+        loadFactor = DEFAULT_LOAD_FACTOR;
+        buckets = new Node[initialCapacity];
+    }
+
+    public MyHashMap(int initialCapacity, float loadFactor) {
+        this.initialCapacity = initialCapacity;
+        this.loadFactor = loadFactor;
+        buckets = new Node[initialCapacity];
+    }
+
+    static class Node<K, V> {
+        final K key;
+        V value;
+        Node<K, V> next;
+
+        public Node(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public int size() {
+        return size;
+    }
+
+    public void put(K key, V value) {
+        int index;
+
+        resize();
+
+        index = indexOf(hash(key));
+
+        Node<K, V> node = new Node<>(key, value);
+
+        if (buckets[index] == null) {
+            buckets[index] = node;
+        } else {
+            Node<K, V> current = buckets[index];
+            while (true) {
+                if (key == null) {
+                    if (current.key == null) {
+                        current.value = value;
+                        return;
+                    }
+                } else if (current.key.equals(key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = node;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+
+        size++;
+
+    }
```

Comment:
1. pls add null check for key(in case when key is null add this node to first bucket)
2. rewrite else case here because 1 case
3. see 2 step